packages/core/src/components/datetime/datetime.tsx

```diff
@@ -0,0 +1,244 @@
+import {
+  DatetimeChangeEventDetail,
+  DatetimeCustomEvent,
+  DatetimeHighlight,
+  DatetimeHighlightCallback,
+  DatetimePresentation,
+} from '@ionic/core'
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Method,
+  Prop,
+  State,
+  h,
+} from '@stencil/core'
+
+@Component({
+  tag: 'atom-datetime',
+  styleUrl: 'datetime.scss',
+  scoped: true,
+})
+export class AtomDatetime {
+  @Element() element!: HTMLElement
+
+  @Prop() label?: string
+  @Prop() cancelText?: string
+  @Prop() clearText?: string
+  @Prop() datetimeId = 'datetime'
+  @Prop() dayValues?: number[] | string
+  @Prop() disabled?: boolean
+  @Prop() doneText?: string
+  @Prop() formatOptions: {
+    date: Intl.DateTimeFormatOptions
+    time: Intl.DateTimeFormatOptions
+  } = {
+    date: {
+      month: '2-digit',
+      day: '2-digit',
+      year: 'numeric',
+    },
+    time: {
+      hour: '2-digit',
+      minute: '2-digit',
+    },
+  }
+  @Prop() highlightedDates?: DatetimeHighlight[] | DatetimeHighlightCallback
+  @Prop() hourCycle: 'h12' | 'h23' = 'h23'
+  @Prop() hourValues?: number[] | string
+  @Prop() locale = 'pt-BR'
+  @Prop() isDateEnabled?: (dateString: string) => boolean
+  @Prop() max?: string
+  @Prop() min?: string
+  @Prop() minuteValues?: number[] | string
+  @Prop() monthValues?: number[] | string
+  @Prop() multiple?: boolean
+  @Prop() name?: string
+  @Prop() preferWheel = false
+  @Prop() presentation?: DatetimePresentation = 'date'
+  @Prop() rangeMode?: boolean
+  @Prop() readonly?: boolean
+  @Prop() showClearButton?: boolean
+  @Prop() showDefaultButtons?: boolean
+  @Prop() showDefaultTimeLabel = false
+  @Prop() showDefaultTitle = false
+  @Prop() useButton = false
+  @Prop() size?: 'cover' | 'fixed' = 'fixed'
+  @Prop({ mutable: true, reflect: true }) value?:
+    | DatetimeCustomEvent
+    | DatetimeChangeEventDetail
+  @Prop() yearValues?: number[] | string
+
+  @State() selectedDates: string[] = []
+
+  @Event() atomFocus!: EventEmitter<void>
+  @Event() atomBlur!: EventEmitter<void>
+  @Event() atomChange!: EventEmitter<string | string[]>
+  @Event() atomCancel!: EventEmitter<string>
+
+  private _datetimeEl!: HTMLIonDatetimeElement
+
+  get datetimeEl(): HTMLIonDatetimeElement {
+    return this._datetimeEl
+  }
+
+  set datetimeEl(value: HTMLIonDatetimeElement) {
+    this._datetimeEl = value
+  }
+
+  @Method()
+  async setValue(value) {
+    if (this.datetimeEl) {
+      this.datetimeEl.value = value
+    }
+  }
+
+  private readonly handleBlur = () => {
+    this.atomBlur.emit()
+  }
+
+  private readonly handleFocus = () => {
+    this.atomFocus.emit()
+  }
+
+  private readonly handleCancel = () => {
+    this.atomCancel.emit()
+  }
+
+  private calculateDateRange(dates: string[]): string[] {
```

Comment:
I didn't understand why it needed this pre-math, to be honest :thinking:  I mean, the date conversion does, but I didn't get the incremental part.

Reply:
This pre-math and increment ensure the function works for both increasing and decreasing date ranges. The increment determines direction, `daysDifference` sets the correct range length, and `setDate` adjusts each step. This guarantees all dates are included, regardless of order

This is the most secure way to avoid errors with some data intervals

Reply:
Got it. I was surprised to see math like that, it looks like something expected from the own component.

Reply:
Ahhh I understand your point now, the problem is because `ion-datetime` [supports multiple date selection] but doesn't have a range of dates 😢 , then we would create this calc every time that we need to use it, to avoid adding more complexity inside our projects I created this abstraction here

Reply:
hmmm I see this now, you told us. Yeah, man, better to keep it here and avoid bugs on the other apps!